### ISSUE TITLE
Camera live filter for photo and video

### ISSUE THREAD

**Issue by JohannesMilke** [flutter/flutter#49531]
I'm wondering if there are any plans to support live filters on videos and photos by the camera plugin. It would be definitely great!

Librarys such as [photofilters](https://github.com/skkallayath/photofilters) could be supported to apply live filters directly on the camera like apps such as snapchat, etc. are doing.

Or are there any possibilities to it right now somehow?

**Comment by guopeng1994** [comment]
+++++1

**Comment by Nish-anth** [comment]
Yeah we definitely need an upgrade to the official camera plugin. Lots of features yet to be implemented. Makes camera app development a huge pain in Flutter, especially the video segment. Based on today's consumer needs and expectations, live filters have become a necessity. Please do the needful.

**Comment by oliverbytes** [comment]
Waiting....

**Comment by II11II** [comment]
Any news?

**Comment by rajatjc** [comment]
Any updates ?

**Comment by czl1378** [comment]
Any updates?

**Comment by Ritsz123** [comment]
waiting for this since a long time...

**Comment by programmermasrawy** [comment]
waiting for flutter to bring it very sooon

**Comment by silversquall** [comment]
any updates ?

**Comment by murtazaagaz** [comment]
really it's a nightmare to add filters to a live camera. Flutter need to do something for it as the filter has become a necessity for the camera.

**Comment by bayuwijdev** [comment]
waiting...

**Comment by Naveed-dev30** [comment]
waiting ...

**Comment by hrithikm2** [comment]
> really it's a nightmare to add filters to a live camera. Flutter need to do something for it as the filter has become a necessity for the camera.

How to do it otherwise, is there a way in flutter to do so?

**Comment by droplet-js** [comment]
any news update?

**Comment by jpetro416** [comment]
Need this...

**Comment by artur-zhdan** [comment]
Really need this

**Comment by deepak786** [comment]
is there any solution to this? 
Can be any third party SDK.

**Comment by anounman** [comment]
any update ??

**Comment by anounman** [comment]
is it possible using camerawesome package ??

https://pub.dev/packages/camerawesome

**Comment by k00na** [comment]
Anyone solved this?

**Comment by MehulKK** [comment]
Any luck on this?

**Comment by g-apparence** [comment]
For those who wants it
CamerAwesome 1.2.0 now provides some basic filters. 

https://twitter.com/mcflyDev/status/1618202544609853440

github : https://github.com/Apparence-io/CamerAwesome

**Comment by B-o-s-s-e** [comment]
waiting…and…hoping. Any updates?

**Comment by PardeepKumar816** [comment]
anyone find any thrid party library or other implementations?

**Comment by ReAlign** [comment]
waiting...

**Comment by llli0v0** [comment]
waiting…and…hoping

**Comment by llli0v0** [comment]
waiting…and…hoping

**Comment by sunjiaming** [comment]
waiting ... anyone find any thrid party library or other implementations?

**Comment by stuartmorgan-g** [comment]
Retitling to reflect the potential feature here. Adding actual filters is out of scope for the plugin, but what would be potentially in scope would be a system that would allow pluging stream transforms (e.g., from other plugins) into the pipeline on the native code side. This would require non-trivial design work, as it would require defining the interface for each native implementation, and a system—either using Dart+FFI, or a native lookup system [similar to what we do with `webview_flutter`](https://github.com/flutter/packages/blob/699fb4d6af0ba5012492a4a50daa82c0f8fd18ea/packages/webview_flutter/webview_flutter_wkwebview/darwin/webview_flutter_wkwebview/Sources/webview_flutter_wkwebview/WebViewFlutterWKWebViewExternalAPI.swift#L28)—for connecting a transform function to a given camera instance.
